exotations/exotica_core_task_maps/src/avoid_look_at_sphere.cpp

```diff
@@ -0,0 +1,186 @@
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+
+#include <exotica_core/server.h>
+#include <exotica_core_task_maps/avoid_look_at_sphere.h>
+
+REGISTER_TASKMAP_TYPE("AvoidLookAtSphere", exotica::AvoidLookAtSphere);
+
+namespace exotica
+{
+AvoidLookAtSphere::AvoidLookAtSphere() = default;
+AvoidLookAtSphere::~AvoidLookAtSphere() = default;
+
+void AvoidLookAtSphere::AssignScene(ScenePtr scene)
+{
+    scene_ = scene;
+    root_frame_name_ = scene_->GetKinematicTree().GetRootFrameName();
+}
+
+void AvoidLookAtSphere::UpdateAsCostWithoutJacobian(Eigen::VectorXdRefConst x, Eigen::VectorXdRef phi)
+{
+    for (int i = 0; i < n_objects_; ++i)
+    {
+        double frac = Eigen::Map<Eigen::Vector3d>(kinematics[0].Phi(i).p.data).topRows<2>().squaredNorm() / radii2_(i);
+        if (frac < 1.0)
+        {
+            phi(i) = 1.0 - 2.0 * frac + frac * frac;
+        }
+        else
+        {
+            phi(i) = 0.0;
+        }
+    }
+}
+
+void AvoidLookAtSphere::UpdateAsInequalityConstraintWithoutJacobian(Eigen::VectorXdRefConst x, Eigen::VectorXdRef phi)
+{
+    for (int i = 0; i < n_objects_; ++i)
+    {
+        phi(i) = radii2_(i) - Eigen::Map<Eigen::Vector3d>(kinematics[0].Phi(i).p.data).topRows<2>().squaredNorm();
+    }
+}
+
+void AvoidLookAtSphere::UpdateAsCostWithJacobian(Eigen::VectorXdRefConst x, Eigen::VectorXdRef phi, Eigen::MatrixXdRef jacobian)
+{
+    for (int i = 0; i < n_objects_; ++i)
+    {
+        Eigen::Vector2d o = Eigen::Map<Eigen::Vector3d>(kinematics[0].Phi(i).p.data).topRows<2>();
+        double frac = o.squaredNorm() / radii2_(i);
+        if (frac < 1.0)
+        {
+            phi(i) = 1.0 - 2.0 * frac + frac * frac;
+
+            for (int j = 0; j < jacobian.cols(); ++j)
+            {
+                jacobian(i, j) = -4.0 * (1.0 - frac) * kinematics[0].jacobian[i].data.topRows<2>().col(j).dot(o) / radii2_(i);
+            }
+        }
+        else
+        {
+            phi(i) = 0.0;
+        }
+    }
+}
+
+void AvoidLookAtSphere::UpdateAsInequalityConstraintWithJacobian(Eigen::VectorXdRefConst x, Eigen::VectorXdRef phi, Eigen::MatrixXdRef jacobian)
+{
+    for (int i = 0; i < n_objects_; ++i)
+    {
+        Eigen::Vector2d o = Eigen::Map<Eigen::Vector3d>(kinematics[0].Phi(i).p.data).topRows<2>();
+        double d = o.norm();
+        phi(i) = radii2_(i) - d * d;
+
+        for (int j = 0; j < jacobian.cols(); ++j)
+        {
+            jacobian(i, j) = -2.0 * kinematics[0].jacobian[i].data.topRows<2>().col(j).dot(o);
+        }
+    }
+}
+
+void AvoidLookAtSphere::PublishObjectsAsMarkerArray()
+{
+    const ros::Time t = ros::Time::now();
+    visualization_msgs::MarkerArray ma;
+    for (int i = 0; i < n_objects_; ++i)
+    {
+        visualization_msgs::Marker m;
+        m.header.stamp = t;
+        m.header.frame_id = "exotica/" + root_frame_name_;
```

Comment:
If this is the only use, don't store it as a private member but use the getter

Reply:
This is not the only use of `root_frame_name_`